### DISCUSSION TITLE
Configuring WebAssembly with Multiple Proxy Endpoints

### DISCUSSION THREAD

**Discussion by lwmorris067** [MarimerLLC/csla#1783] [Questions] [answered]
My team is developing a Blazor WebAssembly app utilizing Csla. We've developed a NuGet package that retrieves configuration information from a database. These objects were also written utilizing Csla. How do configure the web assembly app to call different data portal endpoint for the different libraries?

Am I going about this wrong?

Any help would be appreciated.

Thanks

**Comment by lwmorris067** [comment]
Of course, the next few searches I do after posting this question led me to http://www.lhotka.net/weblog/CSLANETVersion49NewFeatures.aspx. I'm reading it now and hope this answers my question.

**Reply by rockfordlhotka** [reply]
Let me know if you encounter any issues. This is a relatively new feature, and I'm interested in feedback.

**Comment by lwmorris067** [comment]
So, in the end, I decided it would be best to use the server-side dataportal routing vs having the client call a bunch of different services.

I've done what I think I need to do from what I can glean from your blog post (http://www.lhotka.net/weblog/CSLANETVersion49NewFeatures.aspx).

Unfortunatly, when I click my button, nothing happens. No network activity at all. No errors messages in the console. Nothing.

In an effort to troubleshoot, I've removed the server dataportal's routing info and the DataPortalServerRoutingTags from my root business objects.. If the DataPortalServerRouting tag is present, then nothing happens. If I remove it, things work.

An example business object tag:
[DataPortalServerRoutingTag("search-")]

An example route I'm using in the client's server's dataportal constructor:
RoutingTagUrls.Add("search-", "/search-api/dataportal")

The client blazor app has nothing special in it.

Any suggestions?

Thanks for any help.

**Comment by lwmorris067** [comment]
I figured it out. The article had "routingTag- - a routing tag is specified, but no version". I took that to mean I needed to put "search-" in the DataPortalServerRoutingTag. I took out the dash and things started working.

I wish I would have seen an exception of some sort.

**Comment by lwmorris067** [comment]
I'm sorry to bug you again. Do you have an example of two services running in k8s with a blazor webassembly calling a dataportal that's routing to another service's dataportal?

**Comment by lwmorris067** [comment]
It doesn't seem to be routing :( Any help would be appreciated.

**Comment by lwmorris067** [comment]
The controller that is being called by my blazor webassembly client looks like:

    [Route("api/[controller]")]
    [ApiController]
    public class DataPortalController : Csla.Server.Hosts.HttpPortalController
    {
        public DataPortalController()
        {
            UseTextSerialization = true;

            if(!RoutingTagUrls.ContainsKey("search")) RoutingTagUrls.Add("search", "http://search/api/dataportal");
        }

        [HttpGet]
        public string Get()
        {
            return "Running";
        }

        protected override Task PostAsync(string operation, string routingTag)
        {
            Console.WriteLine($"PostAsync - {DateTime.Now} - {operation} - {routingTag}");
            return base.PostAsync(operation, routingTag);
        }

        public override Task PostAsync([FromQuery] string operation)
        {
            Console.WriteLine($"PostAsync - {DateTime.Now} - {operation}");
            return base.PostAsync(operation);
        }
    }

The Root object I'm trying to call has been affixed with 
[DataPortalServerRoutingTag("search")]

**Comment by lwmorris067** [comment]
When I look at the output from the Console.Writes, I see:

PostAsync - 10/6/2020 9:03:59 PM - 'fetch/search-'
PostAsync - 10/6/2020 9:03:59 PM - 'fetch' - 'search-'
PostAsync - 10/6/2020 9:03:59 PM - 'fetch'

for the client data portal. The Search dataportal is never called.

**Comment by lwmorris067** [comment, marked as answer]
I'm so glad I can look at the code!

Turns out the RoutingTagUrls needed a dash in the key:

I had 
RoutingTagUrls.Add("search", "http://search/api/dataportal"); 
vs
RoutingTagUrls.Add("search-", "http://search/api/dataportal");

I also placed the RoutingTagUrls in a static 'constructor' vs the instance constructor.

Thanks for listening!

**Reply by rockfordlhotka** [reply]
Thank you for the commentary on your experience. I've been very busy, so haven't had time to read or reply regularly.

**Reply by rockfordlhotka** [reply]
I've added a work item to provide a better experience around validation of the routing tag values.

https://github.com/MarimerLLC/csla/issues/1858
